samplelistener/listener/dispatcher_service.bal

```diff
@@ -2,46 +2,45 @@ import ballerina/http;
 import ballerina/jballerina.java;
 
 service class DispatcherService {
-   private SlackAppMentionHandlingService|SlackAppCreatedHandlingService serviceRef;
-   //private string verificationToken;
+     private map<GenericService> services;
 
-   isolated function init(SlackAppMentionHandlingService|SlackAppCreatedHandlingService serviceRef) returns error? { //Receive verification token
-        self.serviceRef = serviceRef;
-        //self.verificationToken = "9asdbas9009123nas1e2";
+   isolated function init() returns error? {
+        self.services = {};
    }
 
-   isolated resource function post events (http:Caller caller, http:Request request) returns error? {
-        json payload = check request.getJsonPayload();
-        // string eventOrVerification = check payload.'type;
-
-        // if (payload.token !== self.verificationToken) {
-        //     return error("Verification token mismatch");
-        // }
+   isolated function addServiceRef(string serviceType, GenericService genericService) {
+        self.services[serviceType] = genericService;
+   }
 
-        // if (eventOrVerification == URL_VERIFICATION) {
-        //     check self.verifyURL(caller, payload);
-        // } else if (eventOrVerification == EVENT_CALLBACK) {
-        // }
+   resource function post events (http:Caller caller, http:Request request) returns error? {
+        json payload = check request.getJsonPayload();
 
         GenericEventWrapperEvent genericEvent = check payload.cloneWithType(GenericEventWrapperEvent);
-        if (genericEvent.event.'type == "app_mention") {
-                SlackAppMentionHandlingService serviceReference = <SlackAppMentionHandlingService> self.serviceRef;
-                var s = check self.callOnAppEvent(genericEvent, "app_mention", "onAppMention", serviceReference);
+        match genericEvent.event.'type {
+             "app_mention_added" => {
+                if self.services.hasKey("AppMentionHandlingService") {
+                    AppMentionHandlingService serviceReference = <AppMentionHandlingService> self.services["AppMentionHandlingService"];
+                    check self.callOnAppEvent(genericEvent, genericEvent.event.'type, "onAppMentionAdded", serviceReference);
```

Comment:
The casts such as that in L22 will panic in case the value is of a different service type. But given that we are validating the service type before adding it and the `hasKey` check is done, I would use `map:get()` here.

```suggestion
                    check self.callOnAppEvent(genericEvent, genericEvent.event.'type, "onAppMentionAdded", self.services.get("AppMentionHandlingService"));
```

Reply:
Resolved